### PR TITLE
[DEV-863]: rewrite URLs if they point to https://docs.pagopa.it

### PR DESCRIPTION
**Pull request by MikeAtUqido**
#### List of Changes

* Rewrite URLs if they point to https://docs.pagopa.it pages

#### Motivation and Context

We need to point only on our pages

#### How Has This Been Tested?

- [x] Manually

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 1]
```diff
@@ -2,6 +2,15 @@ import { Config } from '@markdoc/markdoc';
...
 
+const getCorrectGuide = (guide: string) => {
+  switch (guide) {
+    case 'manuale-operativo-dei-servizi':
+      return 'manuale-servizi';
+    default:
+      return guide;
+  }
+};
+
```
Comment (MikeAtUqido) [inline packages/gitbook-docs/src/markdoc/attributes.ts:13]: We need to verify if other guides' name changes between docs and our internal URLs

Comment (marcobottaro) [reply packages/gitbook-docs/src/markdoc/attributes.ts:13]: @certevol is going to map space names from docs.pagopa.it to the developer portal

Comment (certevol) [reply packages/gitbook-docs/src/markdoc/attributes.ts:13]: Added the link to the file in "DEV-863. "The main difference is the direct link on docs.pagopa to the version chosen as the main one. In some cases, the manual's name on docs.pagopa includes also the product name, which would be redundant in the developer portal's slug. Others can be aligned.

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 2]
```diff
@@ -53,7 +82,42 @@ export class LinkAttr {
             spacePrefix.pathPrefix
           )
         : value;
-    } else return value;
+    } else if (value) {
```
Comment (MarcoPonchia) [inline packages/gitbook-docs/src/markdoc/attributes.ts:85]: Maybe here can be process only `value` that include `DOCS_URL` or is necessary to parse all links ? 
Something like: 
```
} else if (value.includes(DOCS_URL)) {
```

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 3]
```diff
@@ -53,7 +82,42 @@ export class LinkAttr {
...
+          .split('/')
+          .filter((p) => p !== '');
+        return finalUrl === `/${name}/${rest.join('/')}`;
+      });
+
+      if (guide) {
+        return guide.path;
+      }
+
+      return value;
```
Comment (MarcoPonchia) [inline packages/gitbook-docs/src/markdoc/attributes.ts:117]: i think could be useful to add a test to cover this case

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 4]
```diff
@@ -2,6 +2,35 @@ import { Config } from '@markdoc/markdoc';
 import path from 'path';
 import { ParseContentConfig } from '../parseContent';
 
+const getCorrectGuide = (guide: string) => {
+  switch (guide) {
```
Comment (marcobottaro) [inline packages/gitbook-docs/src/markdoc/attributes.ts:6]: I think that some of the cases mapped in the spreadsheet are missing

**Review by marcobottaro — CHANGES_REQUESTED** [review]
Check the spreadsheet

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 5]
```diff
@@ -2,6 +2,41 @@ import { Config } from '@markdoc/markdoc';
...
+    case 'gestionedeglierrori':
+      return 'errori';
+    case 'dizionario-dei-metadata':
+      return 'metadata';
+    case 'manuale-operativo-di-firma-con-io':
+      return 'manuale-operativo';
+    default:
+      return guide;
+  }
+};
```
Comment (datalek) [inline packages/gitbook-docs/src/markdoc/attributes.ts:38]: Wouldn't it be better to define the mapping when creating a guide? They could be retrieved from the Config, which contains the information injected by the caller.

File: apps/nextjs-website/src/_contents/products.ts [hunk 6]
```diff
@@ -99,4 +99,24 @@ export const spaceToPrefixMap = pipe(
...
+  'kit-di-comunicazione-per-gli-enti': 'kit-comunicazione',
+  'io-come-aderire': 'accordi-adesione',
+  'f.a.q.-per-integratori': 'knowledge-base',
+  'modello-di-integrazione-di-piattaforma-notifiche': 'modello-di-integrazione',
+  'manuale-operativo-back-office-pagopa-ente-creditore': 'manuale-bo-ec',
+  'manuale-bo-pagopa-psp': 'manuale-bo-psp',
+  gestionedeglierrori: 'errori',
+  'dizionario-dei-metadata': 'metadata',
+  'manuale-operativo-di-firma-con-io': 'manuale-operativo',
+};
```
Comment (datalek) [inline apps/nextjs-website/src/_contents/products.ts:120]: I would simplify it like this:

```TypeScript
export const urlRewritesMap = {
  'https://docs.pagopa.it/manuale-operativo-dei-servizi': 'manuale-servizi',
  'https://docs.pagopa.it/kb-enti': 'supporto-agli-enti',
  'https://domain.com/i-can/put/here/any/path': 'and-is-going/to/be-replaced/with-this-one',
  ...
};
```

By doing this, the logic in `gitbook-docs` can be simplified using a straightforward replacement. For any given URL, you can simply replace the left side with the corresponding right side. 

# Example
Given the following url `https://docs.pagopa.it/modello-di-integrazione-di-piattaforma-notifiche/v/v2.0/` and the following rewrites map:
```TypeScript
export const urlRewritesMap = {
  'https://docs.pagopa.it/modello-di-integrazione-di-piattaforma-notifiche/v/': '/send/guides/modello-di-integrazione/`
  ...
};
```
The system will produce the following url `/send/guides/modello-di-integrazione/v2.0/`

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 7]
```diff
@@ -53,7 +58,34 @@ export class LinkAttr {
...
+      const finalUrl = cleanUrl.replace(currentGuide, correctGuide);
+
+      const guides = parseContentConfig.gitBookPagesWithTitle;
+
+      const guide = guides.find((g) => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const [_product, _path, name, _version, ...rest] = g.path
+          .split('/')
+          .filter((p) => p !== '');
+        return finalUrl === `/${name}/${rest.join('/')}`;
```
Comment (datalek) [inline packages/gitbook-docs/src/markdoc/attributes.ts:78]: There are too many assumptions about how the URL is formatted; I would perform a simple replace based on the urlRewritesMap to apply the rewrites.

**Review by MarcoPonchia — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Comment by MikeAtUqido** [comment]
This PR depends on https://github.com/pagopa/developer-portal/pull/507

File: apps/nextjs-website/src/_contents/products.ts [hunk 8]
```diff
@@ -99,4 +99,47 @@ export const spaceToPrefixMap = pipe(
   RA.uniq(Eq.struct({ spaceId: s.Eq, pathPrefix: s.Eq }))
 );
 
+/**
+ * Contains the mapping between the docs.pago.it url and the developer portal url.
```
Comment (kin0992) [inline apps/nextjs-website/src/_contents/products.ts:103]:
```suggestion
 * Contains the mapping between the docs.pagopa.it url and the developer portal url.
```

**Review by kin0992 — CHANGES_REQUESTED** [review]

**Review by datalek — APPROVED** [review]

**Review by kin0992 — APPROVED** [review]

**Pull request merged**